sdk/core/core/inc/az_curl_adapter.h

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_CURL_ADAPTER_H
+#define AZ_CURL_ADAPTER__H
+
+#include <_az_cfg.h>
+#include <az_callback.h>
+#include <az_http_request.h>
+#include <az_span_seq.h>
+#include <az_write_span_iter.h>
+
+#include <curl/curl.h>
+#include <stdlib.h>
+
+#include <_az_cfg_prefix.h>
+
+typedef struct {
+  CURL * p_curl;
+} az_curl;
+
+AZ_INLINE az_result az_curl_init(az_curl * const out) {
```

Comment:
is there any additional validation we should be doing such as requiring a minimum version of libcurl?

Reply:
great question!!   after a little investigation, we are currently using features added in version 7.1
`(CURLOPT_FAILONERROR, CURLOPT_HTTPHEADER, CURLOPT_URL)`
so that's the min version we need to ask for. 

I will create a new issue to make sure about this

tks